From c1e2a87644fd0aca721a3eb5e2e7c359db977e6d Mon Sep 17 00:00:00 2001
From: Naresh Tanniru <ntanniru@codeaurora.org>
Date: Wed, 20 Mar 2019 19:35:03 +0530
Subject: [PATCH 19/37] audio: Add support for audio extended codecs

- Add support for APTX adaptive and CELT
  formats

[wight554: updated code from LA.QSSI.11.0.r1-05600-qssi.0]

 Conflicts:
	media/java/android/media/AudioSystem.java
	services/core/java/com/android/server/audio/BtHelper.java

Change-Id: I735a297acc4d3d4c7f8d2cc9302a13ebcf4e4c3d
CRs-Fixed: 2433041
Signed-off-by: Volodymyr Zhdanov <wight554@gmail.com>
---
 media/java/android/media/AudioSystem.java | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/media/java/android/media/AudioSystem.java b/media/java/android/media/AudioSystem.java
index da52cfef6bc..85e12c083e1 100644
--- a/media/java/android/media/AudioSystem.java
+++ b/media/java/android/media/AudioSystem.java
@@ -198,6 +198,10 @@ public class AudioSystem
     public static final int AUDIO_FORMAT_APTX_HD        = 0x21000000;
     /** @hide */
     public static final int AUDIO_FORMAT_LDAC           = 0x23000000;
+    /** @hide */
+    public static final int AUDIO_FORMAT_CELT           = 0x26000000;
+    /** @hide */
+    public static final int AUDIO_FORMAT_APTX_ADAPTIVE  = 0x27000000;
 
     /** @hide */
     @IntDef(flag = false, prefix = "AUDIO_FORMAT_", value = {
@@ -224,6 +228,9 @@ public class AudioSystem
             case AUDIO_FORMAT_APTX: return BluetoothCodecConfig.SOURCE_CODEC_TYPE_APTX;
             case AUDIO_FORMAT_APTX_HD: return BluetoothCodecConfig.SOURCE_CODEC_TYPE_APTX_HD;
             case AUDIO_FORMAT_LDAC: return BluetoothCodecConfig.SOURCE_CODEC_TYPE_LDAC;
+            case AUDIO_FORMAT_CELT: return BluetoothCodecConfig.SOURCE_CODEC_TYPE_CELT;
+            case AUDIO_FORMAT_APTX_ADAPTIVE:
+                     return BluetoothCodecConfig.SOURCE_CODEC_TYPE_APTX_ADAPTIVE;
             default:
                 Log.e(TAG, "Unknown audio format 0x" + Integer.toHexString(audioFormat)
                         + " for conversion to BT codec");
@@ -249,6 +256,10 @@ public class AudioSystem
                 return AudioSystem.AUDIO_FORMAT_APTX_HD;
             case BluetoothCodecConfig.SOURCE_CODEC_TYPE_LDAC:
                 return AudioSystem.AUDIO_FORMAT_LDAC;
+            case BluetoothCodecConfig.SOURCE_CODEC_TYPE_CELT:
+                return AudioSystem.AUDIO_FORMAT_CELT;
+            case BluetoothCodecConfig.SOURCE_CODEC_TYPE_APTX_ADAPTIVE:
+                return AudioSystem.AUDIO_FORMAT_APTX_ADAPTIVE;
             default:
                 Log.e(TAG, "Unknown BT codec 0x" + Integer.toHexString(btCodec)
                         + " for conversion to audio format");
-- 
2.28.0

